smithy-aws-traits/src/main/java/software/amazon/smithy/aws/traits/tagging/ServiceTaggingApiValidator.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.aws.traits.tagging;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.shapes.ServiceShape;
+import software.amazon.smithy.model.shapes.ShapeId;
+import software.amazon.smithy.model.validation.AbstractValidator;
+import software.amazon.smithy.model.validation.ValidationEvent;
+
+/**
+ * Validates service satisfies AWS tagging requirements.
+ */
+public final class ServiceTaggingApiValidator extends AbstractValidator {
```

Comment:
> There are a few validators that key off of `@tagEnabled`, are they combinable?

I intentionally separated them out so suppressions could be specific in ignoring certain validation. I don't see a scenario where ignoring all of it would be desirable or useful.

> For naming, I think `Api` can be removed from this.

Changed